### PR TITLE
generate archive name regardless of analysis parent container

### PR DESCRIPTION
**Pull request by kaleb-flywheel**

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -296,16 +296,80 @@ def zip_most_outputs(directory, archive_name, promote_list):
...
-    :param fw_client: the flywheel client
-    :param analysis_id: analysis ID from the config/context
-    :return:
+    A reasonably robust tool for attempting to generate a unique file name for a flywheel analysis. It will try to
+    use the subject code(s) for the input files as the name, and if it cannot, it will fall back on an optionally
+    provided name_string and timestamp. Timestamp will be added whenever both name_str and subject codes are absent
+    :param fw_client: (flywheel.client.Client) An instance of the flywheel client
+    :param config_dict: (dict) a dictionary representation of config.json
+    :param extension: (str) the file extension
+    :param name_string: (str) optional additional label to add to the file label
```
Comment (lmperry) [inline run.py]: Is `name_string` to be provided via config option in the manifest?

File: run.py [hunk 2]
```diff
@@ -320,6 +384,11 @@ def get_subject_code(fw_client, analysis_id):
 
         # Get config options
         config = gear_context.config
+        # Get config.json
```
Comment (lmperry) [inline run.py]: Given the use of gear context, do we need to load data from `config.json` into `config_json`?

Comment (kaleb-flywheel) [reply run.py]: The issue is that there does not appear to be a way to get all inputs regardless of key, given the specificity of gear_context.get_input(). I could use keys specific to siena, but when I was writing the function, my hope was for it to be adaptable to most gears,  regardless of inputs

File: run.py (hunk 1 above)
Comment (kaleb-flywheel) [reply run.py]: name_string is mostly there to allow configuration if I want to copypasta into other gears. I use it to distinguish between the report.siena(x), report.viena, and report.html

**Pull request merged**